sa/model.go

```diff
@@ -1294,3 +1294,82 @@ func setReplacementOrderFinalized(ctx context.Context, db db.Execer, orderID int
 	}
 	return nil
 }
+
+type identifierModel struct {
+	Type  uint8  `db:"identifierType"`
+	Value string `db:"identifierValue"`
+}
+
+func newIdentifierModelFromPB(pb *sapb.Identifier) (identifierModel, error) {
+	idType, ok := identifierTypeToUint[pb.Type]
+	if !ok {
+		return identifierModel{}, fmt.Errorf("unsupported identifier type %q", pb.Type)
+	}
+
+	return identifierModel{
+		Type:  idType,
+		Value: pb.Value,
+	}, nil
+}
+
+func newPBFromIdentifierModel(id identifierModel) (*sapb.Identifier, error) {
+	idType, ok := uintToIdentifierType[id.Type]
+	if !ok {
+		return nil, fmt.Errorf("unsupported identifier type %d", id.Type)
+	}
+
+	return &sapb.Identifier{
+		Type:  idType,
+		Value: id.Value,
+	}, nil
+}
+
+type identifierModels []identifierModel
+
+func newIdentifierModelsFromPB(pb []*sapb.Identifier) (identifierModels, error) {
+	var ids identifierModels
+	for _, p := range pb {
+		id, err := newIdentifierModelFromPB(p)
+		if err != nil {
+			return nil, err
+		}
+		ids = append(ids, id)
+	}
+	return ids, nil
+}
+
+func newPBFromIdentifierModels(ids identifierModels) (*sapb.Identifiers, error) {
+	var pb []*sapb.Identifier
+	for _, id := range ids {
+		p, err := newPBFromIdentifierModel(id)
+		if err != nil {
+			return nil, err
+		}
+		pb = append(pb, p)
+	}
+	return &sapb.Identifiers{Identifiers: pb}, nil
+}
+
+// identifierValuesForType returns a list of identifier values for the given
+// identifier type. If no values are found, an empty list is returned.
+func (id identifierModels) identifierValuesForType(identifierType uint8) []string {
```

Comment:
this method name "stutters" slightly; suggestion:
```suggestion
func (id identifierModels) valuesOfType(identifierType uint8) []string {
```